contracts/src/kleros-v1/KlerosV1Governor.sol

```diff
@@ -0,0 +1,190 @@
+import "./ITokenController.sol";
+import "../arbitration/IArbitrable.sol";
+import "../arbitration/IArbitrator.sol";
+
+/**
+ * @title ERC20 interface
+ */
+interface IPinakion {
+    function balanceOf(address who) external view returns (uint256);
+}
+
+contract KlerosV1Governor is IArbitrable, ITokenController {
+    struct DisputeData {
+        uint256 klerosLiquidDisputeID;
+        bool ruled;
+    }
+
+    IArbitrator public immutable foreignGateway;
+    IKlerosLiquid public immutable klerosLiquid;
+    address public governor;
+
+    mapping(uint256 => uint256) public klerosLiquidDisputeIDtoGatewayDisputeID;
+    mapping(uint256 => DisputeData) public disputes; // disputes[gatewayDisputeID]
+    mapping(address => uint256) public frozenTokens; // frozenTokens[account] locked token which shouldn't have been blocked.
+    mapping(uint256 => mapping(uint256 => bool)) public isDisputeNotified; // isDisputeNotified[disputeID][roundID] used to track the notification of frozen tokens.
+
+    modifier onlyByGovernor() {
+        require(governor == msg.sender);
+        _;
+    }
+
+    /** @dev Constructor. Before this contract is made the new governor of KlerosLiquid, the evidence period of all subcourts has to be set to uint(-1).
+     *  @param _klerosLiquid The trusted arbitrator to resolve potential disputes.
+     *  @param _governor The trusted governor of the contract.
+     *  @param _foreignGateway The trusted gateway that acts as an arbitrator, relaying disputes to v2.
+     */
+    constructor(
+        IKlerosLiquid _klerosLiquid,
+        address _governor,
+        IArbitrator _foreignGateway
+    ) {
+        klerosLiquid = _klerosLiquid;
+        governor = _governor;
+        foreignGateway = _foreignGateway;
+    }
+
+    /** @dev Lets the governor call anything on behalf of the contract.
+     *  @param _destination The destination of the call.
+     *  @param _amount The value sent with the call.
+     *  @param _data The data sent with the call.
+     */
+    function executeGovernorProposal(
+        address _destination,
+        uint256 _amount,
+        bytes calldata _data
+    ) external onlyByGovernor {
+        (bool success, ) = _destination.call{value: _amount}(_data); // solium-disable-line security/no-call-value
+        require(success, "Call execution failed.");
+    }
+
+    /** @dev Changes the `governor` storage variable.
+     *  @param _governor The new value for the `governor` storage variable.
+     */
+    function changeGovernor(address _governor) external onlyByGovernor {
+        governor = _governor;
+    }
+
+    /** @dev Relays disputes from KlerosLiquid to Kleros v2. Only disputes in the evidence period of the initial round can be realyed.
+     *  @param _disputeID The ID of the dispute as defined in KlerosLiquid.
+     */
+    function relayDispute(uint256 _disputeID) external {
+        require(klerosLiquidDisputeIDtoGatewayDisputeID[_disputeID] == 0, "Dispute already relayed");
```

Comment:
I don't see any specification where we don't allow disputeID to be zero. The current KlerosLiquid has a dispute with id 0: https://court.kleros.io/cases/0

This is the same behaviour that I have in the foreign gateway where disputeID starts from 0 and increments.

I can modify the gateway contract to start the disputeID with 1 but is this really required?
(This contract could instead use a boolean mapping)

Reply:
See L82: if the dispute is not in Period.evidence it reverts. So dispute id 0 will not be relayed.

Reply:
Right, there is no specification about this. The assumption that dispute ID can't be zero comes in handy to save storage. I think it would be nice to see this in the foreign gateway and maybe even add it to the standard.

Reply:
Okay, I've updated the foreign gateway contracts to increment the dispute ID from one. That should be enough for now, unless we see any other issues.